### DISCUSSION TITLE
Unable to follow aws instructions for moving an alias in CloudFront due to 'Must specify at least one domain name to use a certificate with a distribution'

### DISCUSSION THREAD

**Discussion by lexhl** [aws/aws-cdk#29947] [Q&A]
Hi, I am trying to follow the instructions on how to move an alias from an existing alias from one CloudFront distribution to another. The instructions says that I need to set up the target distribution with a certificate that includes the alternative domain name that I want to move: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/CNAMEs.html#alternate-domain-names-move-create-target

However when I try to set up the distribution with the certificate but without the alternative domain names (as directed in the instructions) I run into an error `'Must specify at least one domain name to use a certificate with a distribution'` generated I think from this like of code within CDK:
https://github.com/aws/aws-cdk/blob/6fdc4582f659549021a64a4d676fce12fc241715/packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts#L323

It seems that CDK prevents me from adding a certificate unless I have a domain name; but that means I cannot follow the process for moving an alias from and old distribution to a new one.

**Comment by DaniDipp** [comment]
I'm running into the same issue.

For conflicting docs, see
https://github.com/aws/aws-cdk/blob/47605c27fa679b2df98cc73b6b702cc4e4eb75cf/packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts#L134
and
https://github.com/aws/aws-cdk/blob/47605c27fa679b2df98cc73b6b702cc4e4eb75cf/packages/aws-cdk-lib/aws-cloudfront/README.md?plain=1#L118-L126
